packages/react-instantsearch-nextjs/src/InitializePromise.tsx

```diff
@@ -17,6 +19,16 @@ export function InitializePromise() {
       throw new Error('Missing ServerInsertedHTMLContext');
     });
 
+  // Extract search parameters from the search client to use them
+  // later during hydration.
+  let requestParamsList: Array<SearchOptions | undefined>;
```

Comment:
params is always defined, no? why isn't this:

```suggestion
  let requestParamsList: SearchOptions[];
```

Reply:
`params` is typed as an optional parameter

Reply:
right, but when is it actually undefined? isn't it always there? is that to avoid a BC on internal API?

Reply:
This type is optional because you could send manual requests with the client, where queries don't require `params` to be set (https://www.algolia.com/doc/api-reference/api-methods/multiple-queries/#method-param-query-object).

In practice with InstantSearch, the helper always build the requests queries with params.

Reply:
Fine to be extra wary I guess :)